File: public/docs/_examples/template-syntax/ts/app/sizer.component.ts
// #docregion
import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'my-sizer',
  template: `
  <div>
    <button (click)="dec()" title="smaller">-</button>
    <button (click)="inc()" title="bigger">+</button>
    <label [style.font-size.px]="size">FontSize: {{size}}px</label>
  </div>`
})
export class SizerComponent {
  @Input()  size: number;
  @Output() sizeChange = new EventEmitter<number>();

  dec() { this.resize(-1); }
  inc() { this.resize(+1); }

  resize(delta: number) {
    const size = +this.size + delta;
    this.size = Math.min(40, Math.max(8, size));
    this.sizeChange.emit(this.size);
  }
}

File: public/docs/_examples/template-syntax/dart/lib/sizer_component.dart
// #docregion
import 'dart:math';
import 'package:angular2/core.dart';

@Component(
    selector: 'my-sizer',
    template: '''
      <div>
        <button (click)="dec()" title="smaller">-</button>
        <button (click)="inc()" title="bigger">+</button>
        <label [style.font-size.px]="size">FontSize: {{size}}px</label>
      </div>''')
class MySizerComponent {
  static final defaultPxSize = 14;

  @Input()
  String size;

  @Output()
  var sizeChange = new EventEmitter<String>();

  void dec() => resize(-1);
  void inc() => resize(1);

  void resize(num delta) {
    final numSize = num.parse(size, (_) => defaultPxSize);
    size = min(40, max(8, numSize + delta)).toString();
    sizeChange.emit(size);
  }
}
